src/Terrain/GameMap.java

```diff
@@ -1,256 +1,72 @@
 package Terrain;
 
+import Engine.Path;
 import Engine.XYCoord;
 import Units.Unit;
 
-public class GameMap
+public abstract class GameMap
```

Comment:
All right, obtuse question time. Why did we have to split this in three and make it abstract? Couldn't we just keep the functionality here, not have a separate `MapMaster` class, and still have `MapWindow` inherit?

Reply:
Strictly speaking, you totally could.
MapWindows would have a map-ful of Locations that never gets touched, but that's not really a problem.
As I noted up above though, I like having the ability to require a truth-telling map.